codegen/src/test/function.rs

```diff
@@ -252,8 +255,10 @@ mod generate_tests {
                 (actual_diff, expected_diff)
             };
             eprintln!("actual != expected, diverge at char {}", counter);
+            eprintln!("  actual: {}", actual_diff);
```

Comment:
Believe it or not, I would prefer that you delete the entire logic building `actual_diff` and `expected_diff` if this is to get rid of an unused code warning.

I put this in originally, but haven't used it in a long time. It is more visually difficult for me to look at than I thought. It is easier for me to see the character count, and then read the mostly aligned `assert_eq` output, counting the characters from the start.

If you'd like to see what I mean, make a test fail on purpose.

Reply:
No problem.  I can do that.

It was just a matter of preference.  I found an entire screenful of diffs, especially for vary large ones like the deeply-nested test, hard to find the error position, while the 50-character display shows it clearly and concisely.

I originally turned it back on to let me figure out where I went wrong with the last, deep-nesting test.  I suppose it is not very necessary for shorter token streams.

I'll put this under a comment block so that we can simply turn it back on later if needed.